segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/rocksdb/RocksDBCache.java

```diff
@@ -46,6 +47,31 @@
      */
     private static final int MAX_WRITE_AHEAD_LOG_SIZE_MB = 64;
 
+    /**
+     * Max number of in-memory write buffers (memtables) for the cache (active and immutable).
+     */
+    private static final int MAX_WRITE_BUFFER_NUMBER = 4;
```

Comment:
Why do we need 4 memtables?

Reply:
This configuration regarding the number of write buffers is similar to the one proposed here: https://github.com/facebook/rocksdb/wiki/RocksDB-Tuning-Guide
```
max_write_buffer_number = 5;
min_write_buffer_number_to_merge = 2;
```
The default configuration also seems to keep the `2:1` proportion between `max_write_buffer_number` and `min_write_buffer_number_to_merge`.

The intention here is to have some memtables receiving concurrent writes while others are being flushed to disk without causing write stalls.